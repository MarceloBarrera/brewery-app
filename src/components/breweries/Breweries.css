.container {
  display: grid;
  grid-template-columns: 0.1fr 0.9fr;
  /*   grid-template-rows: auto 25px auto 25px auto 25px auto 25px auto 25px auto; */
  grid-column-gap: 10px;
  grid-row-gap: 15px;
  padding: 50px;
}

.grid {
  background-color: white;
  color: #545454;
  padding: 25px;
}

.header {
  grid-column: 1 / 4;
  grid-row: 1 / 2;
}

.header__inner {
  display: flex;
  justify-content: space-between;
  margin-right: -50px;
}
.header__logo img {
  width: 180px;
  height: 20px;
}
.header__logo span {
  margin-left: -20px;
  padding: 2px;
}

.header__refresh-button {
  border-radius: 15px 0 0 15px;
}

.list-buttons {
  display: flex;
  flex: 1;
  justify-content: flex-end;
}

.logo {
  grid-column: 1 / 1;
  grid-row-start: 2 / 3;
  min-height: 300px;
}

.logo img {
  width: 533px;
  height: 694px;
}

.content {
  grid-column: 2 / 4;
  grid-row: 2 / 4;
  min-height: 500px;
  background-color: white;
  margin-left: -30px;
}
