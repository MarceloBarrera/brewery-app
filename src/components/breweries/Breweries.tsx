import React, { useEffect, useReducer } from "react";
import API from "../../Api";

import BreweriesList from "../breweriesList/BreweriesList";
import Button from "../buttons/Button";
// import * as Icons from "./Icons";

import {
  breweriesReducer,
  initialState,
  ActionTypes,
} from "../breweries/BreweriesReducer";
// import logo from "../../assets/some-logo.png";
import "./Breweries.css";

const Breweries = () => {
  const [state, dispatch] = useReducer(breweriesReducer, {
    ...initialState,
  });

  const loadData = React.useCallback(async () => {
    try {
      dispatch({ type: ActionTypes.SET_START_FETCHING });
      const breweries = await API.getBreweries();
      dispatch({ type: ActionTypes.SET_BREWERIES_LIST, payload: breweries });
      dispatch({ type: ActionTypes.SET_END_FETCHING });
    } catch {
      dispatch({ type: ActionTypes.FETCHING_FAILED });
    }
  }, []);

  const sort = () => {
    dispatch({
      type: ActionTypes.SORT_BY_NAME,
    });
  };

  const sortByState = () => {
    dispatch({
      type: ActionTypes.SORT_BY_STATE,
    });
  };

  useEffect(() => {
    (async function () {
      loadData();
    })();
  }, [loadData]);

  return (
    <div className="container">
      <div className="grid header">
        <div className="header__inner">
          <div className="header__logo">
            {/* <img src={logo} alt="logo brewery"></img> */}
            <span>BREWERIES</span>
          </div>
          <Button className="header__refresh-button" onClick={loadData}>
            Reload Data
            {/* <Icons.Refresh /> */}
          </Button>
        </div>
      </div>

      <div className="grid content">
        <div className="list-buttons">
          <Button onClick={sort}>
            Sort by name {state.orderAsc ? "Descending" : "Ascending"}
            {/* <Icons.Sort /> */}
          </Button>

          <Button onClick={sortByState}>
            Sort by state {state.orderAscByState ? "Descending" : "Ascending"}
            {/* <Icons.Sort /> */}
          </Button>
        </div>
        {state.isFetchingBreweries ? (
          <div>Fetching...</div>
        ) : (
          <BreweriesList
            breweries={state.breweriesList}
            errorOcurred={state.errorOccurredWhenFetching}
          ></BreweriesList>
        )}
        {state.errorOccurredWhenFetching ? (
          <div>
            Error occurred when Fetching breweries, please try again later...
          </div>
        ) : (
          ""
        )}
      </div>
    </div>
  );
};

export default Breweries;
