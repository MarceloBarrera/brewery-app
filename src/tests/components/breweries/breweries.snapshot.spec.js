import React from "react";
import renderer from "react-test-renderer";

import Breweries from "../../../components/breweries/Breweries";

it("renders correctly when no fetching data", () => {
  const tree = renderer.create(<Breweries fetchData={false} />);
  expect(tree).toMatchSnapshot();
});
