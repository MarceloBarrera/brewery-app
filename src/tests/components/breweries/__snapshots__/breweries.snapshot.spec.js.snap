// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when no fetching data 1`] = `
<div
  className="container"
>
  <div
    className="grid header"
  >
    <div
      className="header__inner"
    >
      <div
        className="header__logo"
      >
        <img
          alt="logo brewery"
          src="brewery-logo.svg"
        />
        <span>
          BREWERIES
        </span>
      </div>
      <button
        className="header__refresh-button button"
        onClick={[Function]}
      >
        Reload Data
        <img
          alt="icon refresh"
          src="refresh.png"
        />
      </button>
    </div>
  </div>
  <div
    className="grid content"
  >
    <div
      className="list-buttons"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        Sort by type 
        Asc
        <img
          alt="icon sort"
          src="sort.png"
        />
      </button>
      <button
        className="button"
        onClick={[Function]}
      >
        Sort by name 
        Desc
        <img
          alt="icon sort"
          src="sort.png"
        />
      </button>
      <button
        className="button"
        onClick={[Function]}
      >
        Sort by state 
        Asc
        <img
          alt="icon sort"
          src="sort.png"
        />
      </button>
    </div>
    <ul
      className="breweries-list"
    >
      No Breweries yet
    </ul>
    
  </div>
</div>
`;
